ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -1165,4 +1164,72 @@ extension SignalProducerType {
 	public func wait() -> Result<(), Error> {
 		return then(SignalProducer<(), Error>(value: ())).last() ?? .Success(())
 	}
+
+	/// Creates a new `SignalProducer` that will multicast values emitted by
+	/// the underlying producer, up to `capacity`.
+	///
+	/// The underlying `SignalProducer` will not be started until `self` is started
+	/// for the first time. When subscribing to this producer, all previous values
+	/// (up to `capacity`) will be emitted, followed by any new values.
+	///
+	/// If you find yourself needing *the current value* (the last buffered value)
+	/// you should consider using `PropertyType` instead, which, unlike this operator,
+	/// will guarantee at compile time that there's always a buffered value.
+	///
+	/// This operator has the same semantics as `SignalProducer.buffer`.
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public func replayLazily(capacity: Int = Int.max) -> SignalProducer<Value, Error> {
+		precondition(capacity >= 0, "Invalid capacity: \(capacity)")
+
+		var producer: SignalProducer<Value, Error>?
+		var producerObserver: SignalProducer<Value, Error>.ProducedSignal.Observer?
+
+		let lock = NSLock()
+		lock.name = "org.reactivecocoa.ReactiveCocoa.SignalProducer.replayLazily"
+
+		// This will go "out of scope" when the return `SignalProducer` is deallocated.
+		// This lets us know when we're supposed to dispose the underlying producer.
+		let token = NSObject()
+
+		return SignalProducer { observer, disposable in
+			let initializedProducer: SignalProducer<Value, Error>
+			let initializedObserver: SignalProducer<Value, Error>.ProducedSignal.Observer
+			let shouldStartUnderlyingProducer: Bool
+
+			lock.lock()
+			if let producer = producer, producerObserver = producerObserver {
+				(initializedProducer, initializedObserver) = (producer, producerObserver)
+				shouldStartUnderlyingProducer = false
+			} else {
+				let (producerTemp, observerTemp) = SignalProducer<Value, Error>.buffer(capacity)
```

Comment:
Can we get rid of these temporary variables and assign directly to `producer` and `producerObserver`?